lib/apple_support.bzl

```
@@ -265,6 +265,53 @@ def _action_required_attrs():
         ),
     }
 
+def _platform_constraint_attrs():
+    """Returns a dictionary of all known Apple platform constraints that can be resolved.
+
+    The returned `dict` can be added to the rule's attributes using Skylib's `dicts.add()` method.
+
+    Returns:
+        A `dict` object containing attributes to be added to rule implementations.
+    """
+    return {
+        "_ios_constraint": attr.label(
+            default = Label("@platforms//os:ios"),
+        ),
+        "_macos_constraint": attr.label(
+            default = Label("@platforms//os:macos"),
+        ),
+        "_tvos_constraint": attr.label(
+            default = Label("@platforms//os:tvos"),
+        ),
+        "_visionos_constraint": attr.label(
+            default = Label("@platforms//os:visionos"),
+        ),
+        "_watchos_constraint": attr.label(
+            default = Label("@platforms//os:watchos"),
+        ),
+        "_arm64_constraint": attr.label(
+            default = Label("@platforms//cpu:arm64"),
+        ),
+        "_arm64e_constraint": attr.label(
+            default = Label("@platforms//cpu:arm64e"),
+        ),
+        "_arm64_32_constraint": attr.label(
+            default = Label("@platforms//cpu:arm64_32"),
+        ),
+        "_armv7k_constraint": attr.label(
+            default = Label("@platforms//cpu:armv7k"),
+        ),
+        "_x86_64_constraint": attr.label(
+            default = Label("@platforms//cpu:x86_64"),
+        ),
+        "_apple_device_constraint": attr.label(
+            default = Label("@build_bazel_apple_support//constraints:device"),
```

I think we want this to not include the repo name, to support BzlMod better:
```suggestion
            default = Label("//constraints:device"),
```